test/testPriceRouter/PriceRouter.t.sol

```diff
@@ -0,0 +1,788 @@
+            0,
+            badMinPrice,
+            0,
+            false
+        );
+
+        // Simulate calling startEditAsset.
+        bytes32 editHash = keccak256(abi.encode(USDC, settings, abi.encode(stor)));
+        stdstore
+            .target(address(priceRouter))
+            .sig(priceRouter.assetEditableTimestamp.selector)
+            .with_key(editHash)
+            .checked_write(uint256(1));
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__AssetBelowMinPrice.selector,
+                    address(USDC),
+                    price,
+                    badMinPrice
+                )
+            )
+        );
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(stor), 1e8);
+    }
+
+    /**
+     * @notice All pricing operations go through `_getValueInUSD`, so checking for revert in `addAsset` is sufficient.
+     */
+    function testAssetAboveMaxPrice() external {
+        // Store price of USDC.
+        uint256 price = uint256(IChainlinkAggregator(USDC_USD_FEED).latestAnswer());
+
+        // Add USDC again, but set a bad maxPrice.
+        uint144 badMaxPrice = 0.9e8;
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, USDC_USD_FEED);
+        PriceRouter.ChainlinkDerivativeStorage memory stor = PriceRouter.ChainlinkDerivativeStorage(
+            badMaxPrice,
+            0,
+            0,
+            false
+        );
+
+        // Simulate calling startEditAsset.
+        bytes32 editHash = keccak256(abi.encode(USDC, settings, abi.encode(stor)));
+        stdstore
+            .target(address(priceRouter))
+            .sig(priceRouter.assetEditableTimestamp.selector)
+            .with_key(editHash)
+            .checked_write(uint256(1));
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__AssetAboveMaxPrice.selector,
+                    address(USDC),
+                    price,
+                    badMaxPrice
+                )
+            )
+        );
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(stor), 1e8);
+    }
+
+    function testAssetStalePrice() external {
+        // Store timestamp of USDC.
+        uint256 timestamp = uint256(IChainlinkAggregator(USDC_USD_FEED).latestTimestamp());
+        timestamp = block.timestamp - timestamp;
+
+        // Advance time so that the price becomes stale.
+        vm.warp(block.timestamp + 1 days);
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__StalePrice.selector,
+                    address(USDC),
+                    timestamp + 1 days,
+                    1 days
+                )
+            )
+        );
+        priceRouter.getValue(USDC, 1e6, USDC);
+    }
+
+    function testETHtoUSDPriceFeedIsChecked() external {
```

Comment:
Do you think adding another test checking this same thing against different types of `derivatives` for `baseAsset` vs `quoteAsset`?

Reply:
So this test is useful because if you are checking the price of LINK and have a USD price feed for it, you are only reading the USD price feed. But if you are checking the price of BARN, which only has an ETH price feed, you are reading the BARN price feed, and the ETH-USD price feed. 

We already know through other tests that the main price feed(IE BARN) is checked, but this check confirms that we are also validating the ETH-USD price feed is good.

Reply:
We'll leave comment open, the test is a nice to have where we check different derivatives, although we do check them in an isolated mode through other tests.

This was discussed today.